v-next/hardhat/src/internal/builtin-plugins/network-manager/edr/edr-provider.ts

```diff
@@ -153,6 +153,12 @@ export class EdrProvider extends BaseProvider {
 
     let edrProvider: EdrProvider;
 
+    // The chainType can't be undefined, as it's always set on connect()
+    assertHardhatInvariant(
```

Comment:
Could this not be handled at the type level by "resolving" a `networkConfig` with `chainType?: string` to a `ResolvedNetworkConfig` type with `chainType: string`?

Reply:
The `EdrNetworkConfig` is the resolved version of the config, and `chainType` is optional there. However, the `EdrProvider` is only used within the network manager, and if you look at where `.create` is called, we always pass a `chainType`.